src/nvim/ex_getln.c

```diff
@@ -2306,6 +2313,270 @@ static int empty_pattern(char_u *p)
   return n == 0 || (n >= 2 && p[n - 2] == '\\' && p[n - 1] == '|');
 }
 
+handle_T get_cmdpreview_bufnr(void)
```

Comment:
there is an opportunity for a common prefix in all of these functions. Common prefixes are very helpful for discoverability

```
cmdpreview_get_bufnr
cmdpreview_get_ns
cmdpreview_open_buf
cmdpreview_show
...
```